src/file.cpp

```diff
@@ -336,14 +337,30 @@ namespace libtorrent
 		return int(size);
 	}
 
-#ifdef TORRENT_WINDOWS
-	std::string convert_separators(std::string p)
+	template <typename T>
+	std::string convert_from_native_path(T* s);
```

Comment:
Having this declaration without a definition, will delay the error. Instead of being a compilation error you'll get a link error. it seems unnecessary